src/payment/strategies/affirm/affirm-payment-strategy.ts

```diff
@@ -126,17 +126,19 @@ export default class AffirmPaymentStrategy implements PaymentStrategy {
             shipping: this._getShippingAddress(),
             billing: this._getBillingAddress(),
             items: this._getItems(),
-            discounts: this._getDiscounts(),
             metadata: {
                 shipping_type: consignment.selectedShippingOption.type,
                 mode: 'modal',
+                platform_type: 'BigCommerce',
+                platform_version: '',
+                platform_affirm: '',
             },
+            discounts: this._getDiscounts(),
             order_id: order.orderId ? order.orderId.toString() : '',
-            shipping_amount: order.shippingCostTotal * 100,
-            tax_amount: order.taxTotal * 100,
-            total: order.orderAmount * 100,
+            shipping_amount: order.shippingCostTotalAsInteger,
```

Comment:
You'll need to use `AmountTransformer` to do the dollar to cent conversion. It takes into consideration of the decimal places of the transacting currency.